apm-agent-plugins/apm-apache-httpclient/apm-apache-httpclient4-plugin/src/test/java/co/elastic/apm/agent/httpclient/v4/ApacheHttpAsyncClientInstrumentationTest.java

```diff
@@ -125,5 +134,42 @@ public void testSpanWithIllegalProtocol() throws Exception {
         Assertions.assertThat(reporter.getSpans()).hasSize(1);
     }
 
+    @Test
+    public void testPostBodyCapture() throws IOException, ExecutionException, InterruptedException {
+        doReturn(Collections.singletonList(WildcardMatcher.matchAll()))
+            .when(getConfig().getConfig(WebConfiguration.class)).getCaptureClientRequestContentTypes();
+
+        StringBuilder longString = new StringBuilder();
+        for (int i = 0; i < 200; i++) {
+            longString.append(String.format("line %1$4d\n", i));
+        }
+        HttpPost request = new HttpPost(getBaseUrl() + "/");
+        request.setEntity(new InputStreamEntity(new ByteArrayInputStream(longString.toString().getBytes(StandardCharsets.UTF_8))));
+
+        final CompletableFuture<HttpResponse> responseFuture = new CompletableFuture<>();
+
+        client.execute(request, new FutureCallback<>() {
+            @Override
+            public void completed(HttpResponse result) {
+                responseFuture.complete(result);
+            }
+
+            @Override
+            public void failed(Exception ex) {
+                responseFuture.completeExceptionally(ex);
+            }
+
+            @Override
+            public void cancelled() {
+                responseFuture.cancel(true);
+            }
+        });
+        responseFuture.get();
+
 
+        expectSpan("/")
+            .withRequestBodySatisfying(body -> {
+                // assertThat(body).endsWith("line  101\nline");
```

Comment:
hehe

Reply:
whoopsie, fixed in https://github.com/elastic/apm-agent-java/pull/3692/commits/1181ced5180594e4db715d86bdc1105486131a10